langchain-core/src/callbacks/manager.ts

```diff
@@ -1276,8 +1276,12 @@ export class CallbackManager
         handler = new (handlerClass as any)({});
       }
       if (handler !== undefined) {
-        if (!callbackManager?.handlers.some((h) => h.name === handler!.name)) {
-          callbackManager?.addHandler(handler, inheritable);
+        if (!callbackManager) {
```

Comment:
Since this can potentially create a callback manager - let's move this whole block above where we add the `inheritableTags` and `inheritableMetadata` so that those are added properly